packages/react/src/components/Dropdown/Dropdown.js

```diff
@@ -24,304 +16,263 @@ const defaultItemToString = item => {
-        [`${prefix}--dropdown--light`]: light,
-        [`${prefix}--dropdown--${size}`]: size,
-      });
-    const titleClasses = cx(`${prefix}--label`, {
-      [`${prefix}--label--disabled`]: disabled,
-    });
+  /**
+   * Specify a custom `id`
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * Specify whether you want the inline version of this control
+   */
+  inline: PropTypes.bool,
+
+  /**
+   * Specify if the currently selected value is invalid.
+   */
+  invalid: PropTypes.bool,
+
+  /**
+   * Message which is displayed if the value is invalid.
+   */
+  invalidText: PropTypes.string,
+
+  /**
+   * Helper function passed to downshift that allows the library to render a
+   * given item to a string label. By default, it extracts the `label` field
+   * from a given item to serve as the item label in the list.
+   */
+  itemToString: PropTypes.func,
+
+  /**
+   * Function to render items as custom components instead of strings.
+   * Defaults to null and is overriden by a getter
+   */
+  itemToElement: PropTypes.func,
+
+  /**
+   * `onChange` is a utility for this controlled component to communicate to a
+   * consuming component what kind of internal state changes are occuring.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Generic `label` that will be used as the textual representation of what
+   * this field is for
+   */
+  label: PropTypes.node.isRequired,
+
+  /**
+   * Callback function for translating ListBoxMenuIcon SVG title
+   */
+  translateWithId: PropTypes.func,
+
+  /**
+   * 'aria-label' of the ListBox component.
+   */
+  ariaLabel: PropTypes.string,
 
-    const helperId =
-      !id || !helperText ? undefined : `dropdown-helper-text-${id}`;
-    const labelId = `dropdown-label-${id}`;
-    const fieldLabelId = `dropdown-field-label-${id}`;
+  /**
+   * The dropdown type, `default` or `inline`
+   */
+  type: ListBoxPropTypes.ListBoxType,
 
-    const title = titleText ? (
-      <span className={titleClasses}>{titleText}</span>
-    ) : null;
-    const helperClasses = cx(`${prefix}--form__helper-text`, {
-      [`${prefix}--form__helper-text--disabled`]: disabled,
+  /**
+   * Specify the size of the ListBox. Currently supports either `sm`, `lg` or `xl` as an option.
+   */
+  size: ListBoxPropTypes.ListBoxSize,
+
+  /**
+   * In the case you want to control the dropdown selection entirely.
+   */
+  selectedItem: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
+
+  /**
+   * `true` to use the light version.
+   */
+  light: PropTypes.bool,
+
+  /**
+   * Provide the title text that will be read by a screen reader when
+   * visiting this control
+   */
+  titleText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+
+  /**
+   * Provide helper text that is used alongside the control label for
+   * additional help
+   */
+  helperText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+};
+
+export default function Dropdown(props) {
```

Comment:
Style nit:

```suggestion
function Dropdown(props) {
```

`export default` should come after `propTypes` and `defaultProps`